src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Buffers
+{
+    internal sealed class DefaultArrayPool<T> : ArrayPool<T>
+    {
+        private const int MinimiumArraySize = 16;
+        private ArrayPoolBucket<T>[] _buckets;
+
+        internal DefaultArrayPool(int maxLength, int arraysPerBucket)
+        {
+            if (maxLength <= 0)
+                throw new ArgumentOutOfRangeException("maxLength");
+            if (arraysPerBucket <= 0)
+                throw new ArgumentOutOfRangeException("arraysPerBucket");
```

Comment:
The names being used here in the exception differ from the names in the public surface area.  Maybe these checks should be moved out to that Create method instead?